src/model/Course.java

```diff
@@ -86,42 +95,52 @@ public int getAssignmentNeededGradeTotal() {
 		
 		//loop over all arraylist values
 		for (Assignment assignment :assignments) {
-			totalGrade = totalGrade + assignment.getNeededGrade();
+			//do not add if assignment does not have an actual grade defined
+			if(assignment.getActualGrade() >= 0) {
+				totalGrade = totalGrade + assignment.getNeededGrade();
+			}//end if
 		}//end for loop
 		
 		//return average grade
 		return totalGrade;	
-	}
+	}//end getAssignmentNeededGradeTotal
 	
 	//get current overall course grade in percentage
 	public double getAssignmentOverallGrade() {
 		return this.getAssignmentActualGradeTotal() / this.getAssignmentNeededGradeTotal();
-	}
+	}//end getAssignmentOverallGrade
 		
 	//average all assignments grades within the course
 	public double getAssignmentActualGradeAverage() {
```

Comment:
Great use of my "WeightedGrade" constructor. Makes me feel I did something right, haha/